mathesar/views/api.py

```diff
@@ -21,11 +22,18 @@ def create(self, request):
         return Response(serializer.data, status=status.HTTP_201_CREATED)
 
 
-class TableViewSet(viewsets.ReadOnlyModelViewSet):
+class TableViewSet(viewsets.GenericViewSet, ListModelMixin, RetrieveModelMixin):
     queryset = Table.objects.all().order_by('-created_at')
     serializer_class = TableSerializer
     pagination_class = DefaultLimitOffsetPagination
 
+    def create(self, request):
+        file_pk = request.data["file_pk"]
```

Comment:
Tables have a reverse relation to data files. I think you should use the key that's used for the reverse relation here so that there's some relation to the model.

You might also want to add data file IDs to the default table serializer.

Reply:
I messed around a bit with validating using the reverse relation, but ended up getting stuck. In particular, when creating a table from a data file (I assume) we don't care about non-`data_file_pk` fields, so the validation logic gets strange. It seems like we could write custom `.validation()` and `.create()` methods that implement the logic we use outside the serializer right now, but I'm not sure that's preferred.

Although, thinking about it now, there might be cases where we want to combine `data_file_pk` with other fields , we could specifying a new name or schema? Unsure what the long term plans for the table creation API is - maybe something for another PR.

Reply:
Now that I think about it, I think it's probably best to specify the name and schema in this API (and remove the schema from the data file model entirely). It made sense to specify the schema in the data file when we were immediately converting it into a table because we needed that information, but now that table creation is a separate step, the name and schema should be specified here.

Reply:
Got it - I'll probably add the reverse link back in and use the model serializer for validation then.